be/db.go

```diff
@@ -280,18 +285,39 @@ func writeReferral(db *sql.DB, key, value string) error {
 	return err
 }
 
-func generateReferral(db *sql.DB, req *GenRefRequest, codeGen func () string) (*GenRefResponse, error) {
+func generateReferral(db *sql.DB, req *GenRefRequest, codeGen func() string) (*GenRefResponse, error) {
 	log.Printf("Generate and store referral code for the user %s", req.Id)
-	refCode := codeGen()
+
+	rows, err := db.Query(`SELECT referral
+		FROM users_refcodes
+		WHERE id = ?`,
+		req.Id)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var refCode string
+	// Take only the first row. Ignore others as duplicates are not expected.
+	if rows.Next() {
+		if err := rows.Scan(&refCode); err != nil {
+			return nil, err
+		}
+		return &GenRefResponse{
+			RefValue: refCode,
+		}, nil
+	}
+
+	refCode = codeGen()
```

Comment:
Why we still generate referral code instead of using referrer id?
That would guarantee single uniqie code per referrer and if you worry about collisions, randomly generated codes can collide too.